optihood/plot_functions.py

```diff
@@ -1,1027 +1,1031 @@
+        "(('shStorage', 'spaceHeatingBus'), 'flow')": "Storage_sh_out",
+        "storage_content": "Storage_content",
+        "(('shSourceBus', 'shStorage'), 'flow')": "Storage_sh_in",
+        "(('spaceHeatingBus', 'spaceHeating'), 'flow')": "SH_direct_to_load",
+        "(('shDemandBus', 'spaceHeatingDemand'), 'flow')": "Demand_sh",
+        "(('CHP', 'shSourceBus'), 'flow')": "CHP_sh",
+        "(('GasBoiler', 'shSourceBus'), 'flow')": "Gas_sh",
+        "(('GasBoiler', 'dhwStorageBus'), 'flow')": "Gas_dhw",
+        "(('HP', 'shSourceBus'), 'flow')": "HP_sh",
+        "(('GWHP', 'shSourceBus'), 'flow')": "GWHP_sh",
+        "(('electricityBus', 'producedElectricity'), 'flow')": "Self_consumption",
+        "(('gridBus', 'gridElectricity'), 'flow')": "Electricity_grid",
+        "(('pv', 'electricityProdBus'), 'flow')": "PV_elec",
+        "(('solarCollector', 'dhwStorageBus'), 'flow')": "SolarCollector",
+        "(('electricityInBus', 'solarCollector'), 'flow')": "SolarCollector",
+        "(('gridElectricity', 'electricityInBus'), 'flow')": "Electricity_grid",
+        "(('producedElectricity', 'electricityInBus'), 'flow')": "Self_consumption",
+        "(('electricityProdBus', 'electricitySource'), 'flow')": "Battery_bypass"
+    }
+    #if optMode == "group":
+    newLegends["(('electricityBus', 'electricityLink'), 'flow')"] = "electricityLink_out"
+    newLegends["(('electricityLink', 'electricityInBus'), 'flow')"] = "electricityLink_in"
+    newLegends["(('spaceHeatingBus', 'shLink'), 'flow')"] = "shLink_in"
+    newLegends["(('shLink', 'shDemandBus'), 'flow')"] = "shLink_out"
+
+    resultFileBasePath = "..\data\Results"
+    resultFileName = f"results{numberOfBuildings}_{plotOptim}_{optMode}.xlsx"  # add the name of the excel file here for which the plots are to be made
+    if not os.path.exists(resultFileBasePath):
+        os.makedirs(resultFileBasePath)
+    resultFilePath = os.path.join(resultFileBasePath, resultFileName)
+
+    createPlot(resultFilePath, plotLevel, plotType, flowType, plotAnnualHorizontalBar, newLegends)
+
+
+if __name__ == '__main__':
+    optMode = "indiv"  # parameter defining whether the results file corresponds to "indiv" or "group" optimization
+    numberOfBuildings = 4
+    plotOptim = 1  # defines the number of the optimization to plot
+    plotLevel = "allMonths"  # permissible values (for energy balance plot): "allMonths" {for all months}
+    # or specific month {"Jan", "Feb", "Mar", etc. three letter abbreviation of the month name}
+    # or specific date {format: YYYY-MM-DD}
+    plotType = "bokeh"  # permissible values: "energy balance", "bokeh"
+    flowType = "space heat"  # permissible values: "all", "electricity", "space heat", "domestic hot water"
+    plotAnnualHorizontalBar = False  # determines whether the annual horizontal bar is plot or not
+    main(optMode, numberOfBuildings, plotOptim, plotLevel, plotType, flowType, plotAnnualHorizontalBar)

```

Comment:
Function 'main' was renamed to 'plot', as stated in the previous comment, please make the required changes here.